src/io/metadata.rs

```diff
@@ -26,10 +25,37 @@ pub struct Metagram {
     _non_exhaustive: (),
 }
 
-/// A buffer for the extra meta data produced by an image decoder.
+/// Hints whether one specific metadatum is requested.
+#[derive(Clone, Copy)]
+pub enum DatumRequested {
+    /// The decoder should skip the datum when encountered.
+    Ignore,
+    /// The decoder should decode, validate, and add the datum.
+    Record,
+}
+
+/// Configure the caller needs of metadata.
+///
+/// Not every piece of metadata can be added for free. For example, reading EXIF profiles is
+/// commonly associated with decoding of additional chunks, validation, and additional resource
+/// usage. All of these can be opt-out by allowing such a customization. Note that, nevertheless,
+/// the absence of metadata is not definitive proof of its absence in the original file. It might
+/// be more recent than the decoder and not yet have support.
+#[derive(Clone, Default)]
+// We may want to add a non-copy field?
+#[allow(missing_copy_implementations)]
+pub struct RecorderConfig {
+    /// Whether the decoder should try to provide color profile data.
+    pub color_profile: DatumRequested,
```

Comment:
Any particular reason this couldn't be `pub skip_color_profile: bool`?

Reply:
Yes, but it took me a while to make up my own mind. My vague idea was to have a third variant, `Required`, that would _fail_ decoding when no information is available. This could allow decoders to fail fast instead of needing to wait for the final check which would complement `Limits` nicely. However, there are some pain points:

* We will have to add a check _after_ the decoder in any case if we want to make the an actionable guarantee for the caller. But when those checks trigger we will merely discard already correct data which is rather odd and mostly detrimental.
* If we have any sort of recovery or fallback then we need to relax it to `Requested` in any case with similar concerns as above.
* There might be overlapping constraints such as 'ensure a correct, complete and accurate record of all color conversion and quantization steps in the image creation pipeline' which wouldn't fit in the tri-state idea in any case.
* They can always be converted to a flag on the recorded `MetadataContainer`. Consider the above then we could add a `bool` attribute that signals the completeness of the color information in other fields.

This leaves only code clarity as a potential benefit of an explicit new enum.